WordPress/gutenberg
# Style engine: export util to compile CSS custom var from preset string

## What?

Follow up on https://github.com/WordPress/gutenberg/pull/64128

This PR:

- Removes `compileStyleValue()` from global style output functions
- Uses the style engine's `getCSSVarFromStyleValue()` in its place
- Update style engine tests to cover bugfix in https://github.com/WordPress/gutenberg/pull/43116 
- Moves style engine utils tests into correct directory

## Why?

`getCSSVarFromStyleValue()` does the same thing as `compileStyleValue()`

This PR ensures that it's more widely available throughout the app as a public API.

## How?

Shuffling stuff around.

## Testing Instructions

There should be no regressions. 

Smoke test in the site editor by applying some global styles presets to block styles, e.g., colors or font sizes. 

Add a custom color in Styles > Color > Palette and make sure the CSS custom var is generated correctly, e.g.,

<img width="274" alt="Screenshot 2024-08-14 at 4 30 44 PM" src="https://github.com/user-attachments/assets/396e7310-175f-40b3-9c73-15296a6715f0">

```css
.has-custom-hot-dog-color {
    color: var(--wp--preset--color--custom-hot-dog) !important;
}
```

Run the tests!

```
npm run test:unit packages/block-editor/src/components/global-styles/test/use-global-styles-output.js
npm run test:unit packages/style-engine/src/styles/test/utils.js
```
